pkg/kademlia/peer_discovery.go

```diff
@@ -83,13 +83,17 @@ func (lookup *peerDiscovery) Run(ctx context.Context) (target *pb.Node, err erro
 						working++
 						break
 					}
-
 					// no work, wait until some other routine inserts into the queue
 					lookup.cond.Wait()
 				}
 				lookup.cond.L.Unlock()
-
-				neighbors, err := lookup.client.Lookup(ctx, *next, pb.Node{Id: lookup.target})
+				var nodeType pb.NodeType
+				if target != nil {
+					nodeType = target.Type
+					nodeType.PanicOnInvalid("Peer Discovery Run")
+				}
+				next.Type.PanicOnInvalid("next")
```

Comment:
@egonelbre it's panicking here during tests but I can't figure out why next.Type is nill